nni/retiarii/graph.py

```diff
@@ -0,0 +1,654 @@
+"""
+Model representation.
+"""
+
+import copy
+from enum import Enum
+import json
+from collections import defaultdict
+from typing import (Any, Dict, List, Optional, Tuple, Union, overload)
+
+from .operation import Cell, Operation, _IOPseudoOperation
+
+__all__ = ['Model', 'ModelStatus', 'Graph', 'Node', 'Edge', 'IllegalGraphError', 'MetricData']
+
+
+MetricData = Any
+"""
+Graph metrics like loss, accuracy, etc.
+
+# Maybe we can assume this is a single float number for first iteration.
+"""
+
+
+class TrainingConfig:
```

Comment:
Please write real doc for this class.
I copied it from artifact code and have no idea how it works.